torch/csrc/jit/serialization/export.cpp

```diff
@@ -583,7 +589,7 @@ void GraphEncoder::EncodeValueInfoType(
       // Encode type if either shape or dtype exists.
       onnx::TypeProto_Tensor* onnx_tensor_type =
           onnx_type->mutable_tensor_type();
-      tensorTypeToONNXType(tensor_type, onnx_tensor_type);
+      tensorTypeToONNXType(tensor_type, onnx_tensor_type, tensor_type != n->type());
```

Comment:
Please move the comment from lines 563-564 to here and expand it a bit to explain how it relates to `tensor_type != n->type()`. That is, make it clear how `tensor_type != n->type()` relates to sequence types.

Reply:
Changing to type check whether `n` is List type.